src/client/dcp_client/gui/napari_window.py

```diff
@@ -22,33 +50,208 @@ def __init__(self, app: Application):
-        buttons_layout.addWidget(add_to_curated_button)
+        layout.addWidget(add_to_curated_button, 2, 2, 1, 2)
         add_to_curated_button.clicked.connect(self.on_add_to_curated_button_clicked)
 
-        layout.addLayout(buttons_layout)
-
         self.setLayout(layout)
-        self.show()
+
+    def switch_controls(self, target_widget, status: bool):
+        getattr(self.qctrl, target_widget).setEnabled(status)
+
+    def switch_to_active_mask(self):
+
+        self.switch_controls("paint_button", True)
+        self.switch_controls("erase_button", True)
+        self.switch_controls("fill_button", False)
+
+        self.active_mask = True
+    
+    def switch_to_non_active_mask(self):
+
+        self.instances = set(np.unique(self.layer.data[self.active_mask_index])[1:])
+
+        self.switch_controls("paint_button", False)
+        self.switch_controls("erase_button", False)
+        self.switch_controls("fill_button", True) 
+
+        self.active_mask = False
+
+    def set_active_mask(self):
+        self.mask_choice_dropdown.setDisabled(True)
+        self.active_mask_index = self.mask_choice_dropdown.currentData()
+        self.instances = set(np.unique(self.layer.data[self.active_mask_index])[1:])
+        self.prev_mask = self.layer.data[self.active_mask]
+        if self.active_mask_index == 1:
+            self.switch_to_non_active_mask()
+
+    def find_edges(self, idx=None):
+        '''
+        idx - indices of the specific labels from which to get contour
+        '''
+        if idx is not None and not isinstance(idx, list):
+            idx = [idx]
+
+        active_mask = self.layer.data[self.active_mask_index]
+
+        instances = np.unique(active_mask)[1:]
+        edges = np.zeros_like(active_mask).astype(int)
+
+        # to merge the discontinuous contours
+        kernel = np.ones((5, 5))
+
+        if len(instances):
+            for i in instances:
+                if idx is None or i in idx:
+        
+                    mask_instance = (active_mask == i).astype(np.uint8)
+
+                    edge_mask = 255 * (canny(255 * (mask_instance)) > 0).astype(np.uint8)
+                    edge_mask = cv2.morphologyEx(
+                        edge_mask, 
+                        cv2.MORPH_CLOSE, 
+                        kernel,
+                    )
+                    edges = edges + edge_mask
+
+            # if masks are intersecting then we want to count it only once
+            edges = edges > 0
+            # cut the contours
+            self.layer.data = self.layer.data * np.invert(edges).astype(np.uint8)
+            
+    def copy_mask_callback(self, layer, event):
+
+        source_mask = layer.data
+    
+        if event.type == "mouse_press":
+
+            c, event_x, event_y = event.position
+            c, event_x, event_y = int(c), int(np.round(event_x)), int(np.round(event_y))
+
+            if source_mask[c, event_x, event_y] == 0:
+                self.new_pixel = True
```

Comment:
you don't need to assign variables which you only call within the function to the class. ```self.new_pixel``` can just be ```new_pixel```

Reply:
actually new_pixel isn't used anywhere so these lines can be removed?